contents/docs/configuring-posthog/scaling-posthog.md

```diff
@@ -8,11 +8,23 @@ A standard instance of PostHog using PostgreSQL is pretty efficient for storing
 
 However, once you have more data, you might start having issues while running queries. The issue is that Postgres isn't very good at doing analytical queries on larger datasets. This is where ClickHouse comes in.
 
+There are quite a few factors than can contribute to whether your Postgres instance is enough for running PostHog, for example:
+- Recurring event ingestion volume (details on that below).
+- Resources (RAM & CPU usage) allocated directly to your Postgres instance.
+- Disk read/write speed. One thing that can really help here is using solid-state disks. Provisioned IOPS (supported with some infrastructure providers, can also help a lot).
+- Number of dashboards (and items on those dashboards that you have). These get recomputed periodically. See [Environment variables](/docs/configuring-posthog/environment-variables) for details on how to adjust these.
+- Number of actions & cohorts that you have defined (these too get recomputed periodically). See [Environment variables](/docs/configuring-posthog/environment-variables) for details on how to adjust these.
+
+
+Event ingestion volume can be one of the largest contributing factors to your database load. This is because the more events you have, the more data that needs to be scanned & analyzed when doing any sort of querying. We're still testing this out and there are other factors to consider, but based on our experience here's a rule of thumb on what event volume runs well on a Postgres backend.
+- At 2M (two million) events per month, Postgres starts to struggle and queries will start taking noticeably longer than expected.
+- 5M events per month seems to be a hard limit on what Postgres can performantly manage. If your volume is higher than this, we recommend you switch to PostHog Cloud or our ClickHouse backend.
```

Comment:
Might be worth clarifying how much you need to pay or what kind of data volume (gb) and db server we are looking at here.

I think we had more volume on cloud, but that thing was hooked to heroku's largest postgres IIRC...

Reply:
Yeah that would be super helpful! Don't have that context though, maybe @macobo? I think you did a test fairly recently right?

Reply:
Basic rule of thumb here is the Postgres instance should have enough memory to store the entire events table in memory. So if you have a 32 gb events table you'll want at least 32 gb of memory on Postgres. This get's $$ quick though.

You can find the table sizes using this
https://wiki.postgresql.org/wiki/Disk_Usage

If I remember correctly 1bn events equates to about 300gb on disk for Postgres (including indexes). The Postgres instance we were using on Heroku had 488gb of ram.

This breaks down to about 300mb of disk and ram needed per million events stored.

Reply:
This is amazing! Thanks for the context @fuziontech ! Added these details, merging now